library/Director/Import/ImportSourceRestApi.php

```diff
@@ -273,11 +273,9 @@ protected static function addProxy(QuickForm $form)
                     ),
                 ]);
 
-                $passRequired = strlen($form->getSentOrObjectSetting('proxy_user')) > 0;
-
                 $form->addElement('storedPassword', 'proxy_pass', [
                     'label'    => $form->translate('Proxy Password'),
-                    'required' => $passRequired
+                    'required' => strlen((string) $form->getSentOrObjectSetting('proxy_user')) > 0
```

Comment:
I think you don't need `strlen`. `$form->getSentOrObjectSetting('proxy_user') ?: false` should be sufficient.

Reply:
I don't think so, that is not a strict comparison.

Reply:
The returned value of the function can be a string, '' or null. And all these cases are covered in the suggested change? Or am I missing something?

Reply:
Did you test it? Try entering '0', for 'proxy_user field'.

Reply:
Ohh, I see, then you can use `??` instead of `?:`.

Reply:
As discussed offline this would not work as well.